tools/registry.py

```diff
@@ -434,6 +435,30 @@ def add(self, module, override=False):
     metadata["versions"].sort(key=Version)
     json_dump(metadata_path, metadata)
 
+  def update_versions(self, module_name):
+    """Update the list of versions in the metadata.json."""
+    metadata = self.get_metadata(module_name)
+    yanked_versions = set(metadata["yanked_versions"])
+    module_path = self.root / "modules" / module_name
+    actual_versions = {v.name for v in module_path.iterdir() if v.is_dir()}
+    metadata["versions"] = sorted(actual_versions - yanked_versions, key=Version)
```

Comment:
I think that yanked versions should also be listed here.

@meteorcloudy Is that true?

Reply:
Yeah, let's keep the yanked versions here for completeness.

Bazel actually doesn't read anything other than the `yanked_version` field: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java;l=393

However, the BCR UI might still use this information to render the page for the module: https://registry.bazel.build/modules/zlib

Reply:
changed it to just be all the dirs regardless of what's yanked

Reply:
Thanks!